File: Algorithms/278. First Bad Version/Solution.java
/* The isBadVersion API is defined in the parent class VersionControl.
      boolean isBadVersion(int version); */

public class Solution extends VersionControl {
    public int firstBadVersion(int n) {
        if(isBadVersion(1)){
            return 1;
        }
        if(!isBadVersion(n-1)){
            return n;
        }
        int s=0;
        int e=n;
        int t=s/2+e/2+(s%2+e%2)/2;
        while(!(!isBadVersion(t)&&isBadVersion(t+1))){
            if(isBadVersion(t)){
                e=t;
            }else{
                s=t;
            }
               t=s/2+e/2+(s%2+e%2)/2;
        }
        return t+1;
    }
}

File: Algorithms/278. First Bad Version/Solution.py
#coding=utf-8
__author__ = 'xuxuan'
# The isBadVersion API is already defined for you.
# @param version, an integer
# @return a bool
def isBadVersion(version):
    return True
class Solution(object):
    def firstBadVersion(self, n):
        """
        :type n: int
        :rtype: int
        """
        s=1
        e=n
        while s!=e:
            m=(s+e)/2
            if isBadVersion(m):
                e=m
            else:
                s=m+1
        return s if isBadVersion(s) else (s+1)